storage/storage-interface/src/lib.rs

```diff
@@ -561,6 +568,42 @@ impl MoveStorage for &dyn DbReader {
             .collect()
     }
 
+    fn fetch_config_by_version(&self, config_id: ConfigID, version: Version) -> Result<Vec<u8>> {
+        let core_storage_account_state = AccountState::try_from(
+            &self
+                .get_account_state_with_proof_by_version(CORE_CODE_ADDRESS, version)?
+                .0
+                .ok_or_else(|| {
+                    format_err!("missing blob in account state/account does not exist")
+                })?,
+        )?;
+
+        match core_storage_account_state.get(&experimental_access_path_for_config(config_id).path) {
+            Some(config) => Ok(config.to_vec()),
+            _ => {
+                let diem_root_state = AccountState::try_from(
+                    &self
+                        .get_account_state_with_proof_by_version(diem_root_address(), version)?
+                        .0
+                        .ok_or_else(|| {
+                            format_err!("missing blob in account state/account does not exist")
+                        })?,
+                )?;
+                diem_root_state
+                    .get(&default_access_path_for_config(config_id).path)
```

Comment:
What would you think of renaming the experimental version so that there wouldn't need to be special-casing here? I think that coupled with my comment above about changing the core configs address to be `diem_root` might make this code cleaner

Reply:
Not sure what "renaming the experimental version" means? Right now experimental DiemVersion is of type `DiemVersion::DiemVersion` while the DPN one is of type `DiemConfig<DiemVersion::DiemVersion>`. Do we want to use DiemConfig for experimental too?

Reply:
Ahhh good point, I misunderstood what was going on here :) I don't think we want/need to use DiemConfig for experimental